app/models/shipit/task.rb

```diff
@@ -236,8 +237,8 @@ def request_abort
       end
     end
 
-    def abort!(rollback_once_aborted: false)
-      update!(rollback_once_aborted: rollback_once_aborted)
+    def abort!(rollback_once_aborted: false, aborted_by:)
+      update!(rollback_once_aborted: rollback_once_aborted, aborted_by_id: aborted_by.id)
```

Comment:
This is being opinionated that aborted_by is always present and a user object. No nil. Which I think is ok. But the column is nulable, meaning that’s only for backwards compatibility. Not going forward.

Reply:
The column needs to be null because tasks can be in any state, for example in the running state, and tasks in states other than aborted, aborting and flapping should not have a aborted _by.